spin/cmds/meson.py

```diff
@@ -36,6 +36,26 @@ def _set_pythonpath(quiet=False):
     site_packages = _get_site_packages()
     env = os.environ
 
+    cfg = get_config()
+    package = cfg.get("tool.spin.package", None)
+    if package:
+        import importlib_metadata
+
+        try:
+            dist = importlib_metadata.Distribution.from_name(package)
+            if dist.origin.dir_info.editable:
```

Comment:
Should this also raise a different warning if a non-editable but installed distribution is detected? Possibly outside the directory expected by spin? That might lead to similar confusion.

Looks like an improvement otherwise! :+1:

Reply:
Fortunately, a non-editable installation does not cause problems.

The editable installs are particularly obnoxious, because their hooks are run before the PYTHONPATH takes effect.